### ISSUE TITLE
Hamcrest 2.0.0.0 does not work with JDK 6

### ISSUE THREAD

**Issue by marcphilipp** [hamcrest/JavaHamcrest#105]
If I read https://github.com/hamcrest/JavaHamcrest/issues/65#issuecomment-68097018 correctly, Hamcrest 2.0.0.0 was supposed to work with JDK 6.

However, it was apparently compiled for JDK 7.

I've create a small sample project at https://github.com/marcphilipp/hamcrest-2-jdk6. It uses Maven to fetch `java-hamcrest` and `hamcrest-java` 2.0.0.0. It's [main class](https://github.com/marcphilipp/hamcrest-2-jdk6/blob/master/src/main/java/de/marcphilipp/hamcrest2/HamcrestTest.java) declares a simple assertion.

You can build a jar-with-dependencies using Maven:

```
$ mvn clean compile assembly:single
```

When executed using a JVM < 1.7 it fails with the following exception:
```
$ /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/bin/java -jar target/hamcrest-2-jdk6-0.0.1-SNAPSHOT-jar-with-dependencies.jar 
Exception in thread "main" java.lang.UnsupportedClassVersionError: org/hamcrest/CoreMatchers : Unsupported major.minor version 51.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClassCond(ClassLoader.java:637)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:621)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:141)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:283)
	at java.net.URLClassLoader.access$000(URLClassLoader.java:58)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:197)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:301)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
	at de.marcphilipp.hamcrest2.HamcrestTest.main(HamcrestTest.java:8)
```

When run using JDK 7 it gives the following (expected!) output:
```
$ /Library/Java/JavaVirtualMachines/1.7.0.jdk/Contents/Home/bin/java -jar target/hamcrest-2-jdk6-0.0.1-SNAPSHOT-jar-with-dependencies.jar
Exception in thread "main" java.lang.AssertionError: 
Expected: a string starting with "1.6.0"
     but: was "1.7.0_04-b20"
	at org.hamcrest.junit.MatcherAssert$1.mismatch(MatcherAssert.java:19)
	at org.hamcrest.junit.internal.Matching.checkMatch(Matching.java:20)
	at org.hamcrest.junit.MatcherAssert.assertThat(MatcherAssert.java:16)
	at org.hamcrest.junit.MatcherAssert.assertThat(MatcherAssert.java:12)
	at de.marcphilipp.hamcrest2.HamcrestTest.main(HamcrestTest.java:9)
```

**Comment by marcphilipp** [comment]
JUnit 5 will depend on JDK 6, not JDK 7... so this issue is really critical for me. :-/

**Comment by npryce** [comment]
Rereading that comment thread, the eventual agreement was that the next release of Hamcrest would be 2.0.0.0 and would support Java 7 and above.

If JUnit 5 supports Java 6 and above, that will be fine if Hamcrest integration is taken out of JUnit and implemented in Hamcrest-JUnit instead.  Users of Hamcrest and Hamcrest-JUnit will be using Java 7+, and so can use JUnit 5, but JUnit 5 users who stick with Java 6 will have to do without Hamcrest.

**Comment by npryce** [comment]
I've released Hamcrest-JUnit 1.0.0.1, which is compiled with JDK 1.6.
